src/vs/workbench/parts/terminal/electron-browser/terminalActions.ts

```diff
@@ -657,22 +657,39 @@ export class RunActiveFileInTerminalAction extends Action {
 		}
 		let uriPath: string = uri.fsPath;
 		const hasSpace = uriPath.indexOf(' ') !== -1;
-		if (hasSpace && isWindows) {
-			uriPath = '"' + uriPath + '"';
-		} else if (!isWindows) {
-			if (uriPath.indexOf('\\') !== 0) {
-				uriPath = uriPath.replace(/\\/g, '\\\\');
+		if (isWindows) {
+			const exe = this.terminalService.getActiveInstance().shellLaunchConfig.executable;
+			const osVersion = (/(\d+)\.(\d+)\.(\d+)/g).exec(os.release());
+			let buildNumber: number = 0;
+			if (osVersion && osVersion.length === 4) {
+				buildNumber = parseInt(osVersion[3]);
 			}
-			const hasDoubleQuote = uriPath.indexOf('"') !== -1;
-			if (!hasSpace && hasDoubleQuote) {
-				uriPath = '\'' + uriPath + '\'';
+			// 17063 is the build number where wsl path was introduced.
+			// Update Windows uriPath to be executed in WSL.
+			if (((exe.indexOf('wsl') !== -1) || ((exe.indexOf('bash.exe') !== -1) && (exe.indexOf('git') === -1))) && (buildNumber >= 17063)) {
+				uriPath = 'runActive="$(wslpath ' + this._escapeNonWindowsPath(uriPath) + ')" && "${runActive}"';
```

Comment:
This can be async as run already returns a promise, we could resolve by shelling out to `wslpath` and send a nicer path to the shell?

Reply:
wslpath can only be run from WSL. I haven't found a better way to do this 😞

Reply:
Can you run `system32/bash.exe`, `['-c', 'wslpath path']`?

Reply:
From cmd, yes, but not using execFile.